pkg/cmd/kind/load/docker-image/docker-image.go

```diff
@@ -128,16 +133,33 @@ func runE(logger log.Logger, flags *flagpole, args []string) error {
 	fns := []func() error{}
 	for i, imageName := range imageNames {
 		imageID := imageIDs[i]
+		processed := false
 		for _, node := range candidateNodes {
-			id, err := nodeutils.ImageID(node, imageName)
-			if err != nil || id != imageID {
-				selectedNodes = append(selectedNodes, node)
-				logger.V(0).Infof("Image: %q with ID %q not yet present on node %q, loading...", imageName, imageID, node.String())
+			exists, reTagRequired := checkIfImageReTagRequired(node, imageID, imageName)
+			if !exists {
+				// This could be because of one of two reasons. Either the previous command ran into an error or no image
+				// with matching Image ID was found. Let us try again and see if the image is indeed required to be loaded
+				id, err := nodeutils.ImageID(node, imageName)
+				if err != nil || id != imageID {
+					selectedNodes = append(selectedNodes, node)
+					logger.V(0).Infof("Image: %q with ID %q not yet present on node %q, loading...", imageName, imageID, node.String())
+				}
+				continue
+			}
+			if reTagRequired {
```

Comment:
go encourages only indenting the exceptional cases, I think it would be more idomatic and readable do to more like:
```go
if exists {
    continue
}
if reTagRequired {
   // do retagging instead of loading
    ...
    continue
}
// do loading
...
```

Reply:
Ack. Let me refactor this.

Reply:
I am going to refactor the code a bit and instead of returning two booleans, going to return an `ENUM` like entity that we can use to compare and make a decision. It will also keep the conditional checks cleaner.  Do you see any issue with this @BenTheElder 

```golang
func checkIfImageReTagRequired(node nodes.Node, imageID, imageName string, tagFetcher imageTagFetcher) processMode {
	tags, err := tagFetcher(node, imageID)
	if len(tags) == 0 || err != nil {
		return LoadImage
	}

	imageName = sanitizeImage(imageName)
	if ok := tags[imageName]; ok {
		return NoOp
	}
	return ReTagImage
}

-----
if imageProcessMode == NoOp {
    continue
}

if imageProcessMode == ReTagImage {
    // do stuff
    continue
}
// load Image
```

Reply:
I wouldn't recommend it here because go enums are weak and we wind up with an unhandled case for additional values we didn't consider.

The booleans are reasonably idiomatic and are clearly exhaustively handled.

Reply:
> I wouldn't recommend it here because go enums are weak

I think i kind of mentioned it wrong.  I meant to say a custom type. Not enum,. 

```golang
type (
	processMode     string
)

const (
	LoadImage  processMode = "load"
	ReTagImage processMode = "re-tag"
	NoOp       processMode = "none"
)
```

Reply:
I don't think this is an idiomatic approach in go, versus two return values, or inlining the method

Reply:
the problem is, even though there are only 3 states, I need to go read this other type and see what values exist to understand if they're all being handled.

With two booleans with local variable names, I can quickly comprehend if it's being handled.

In the future if a fourth mode is added to this type for some reason, subtle bugs will creep in because we're still using a very weak "enum" and it's not obvious if they're all being handled where they need to be, versus I know the booleans only have two values to handle.

Reply:
@BenTheElder  All right. Let me move it back to the two bool return format and add checks accordingly.